mlflow/utils/search_utils.py

```diff
@@ -356,7 +356,7 @@ def _get_value(cls, identifier_type, key, token):
                         "Found {}".format(key, token.value),
                         error_code=INVALID_PARAMETER_VALUE,
                     )
-                return token.value
+                return int(token.value)
```

Comment:
```suggestion
                return float(token.value)
```

?

Reply:
I think the best approach here is split `NUMERIC_ATTRIBUTES` into int and float.